common/reconciliation/common/interfaces.go

```diff
@@ -38,25 +38,25 @@ type (
 	// InvariantManager represents a manager of several invariants.
 	// It can be used to run a group of invariant checks or fixes.
 	InvariantManager interface {
```

Comment:
Should we invert the dependency here? meaning putting Check() and Fix() as execution methods.

Reply:
Or you can continue with the current approach and we can rethink about this when Mantas starts to break this workflow into pieces.

Reply:
Yep, we definitely need that! Will try to do in the second "implement concrete execution check" step.